solver/src/driver.rs

```diff
@@ -414,6 +421,14 @@ impl Driver {
 
             tracing::debug!("winning settlement: {:?}", settlement);
             self.submit_settlement(settlement.clone()).await;
+            inflight_orders = settlement
+                .settlement
+                .settlement
+                .trades()
+                .iter()
+                .map(|t| t.order.order_meta_data.uid)
```

Comment:
We should probably only remove fully filled orders?

Reply:
Although if we do this, we still could run into the same problem trying to "over fill an order". I would think that missing at most one single run isn't so bad in this case.

Reply:
> I would think that missing at most one single run isn't so bad in this case

It's essentially censoring this order and could potentially be used in an attack. Match some order partially as you know it won't be in the system next round. We could also store the remaining sell amount for those orders and update if they differ from what the orderbook sent us.

Reply:
Ok, so I am in the process of making this special consideration for partially fillable orders and stumbled upon an outstanding `TODO` in the code (particularly in `normalize_limit_order` which converts `Order` to `LimitOrder`

http://github.com/gnosis/gp-v2-services/blob/51d10253916e41cba150242cd874237bafb8a745/solver/src/liquidity/offchain_orderbook.rs#L60-L62

we see that it remains to discount previously executed amounts here. The question I would have it whether or not the solver takes the `OrderMetaData` into consideration when working with `LimitOrder`. 

More specifically if it would be a problem for settlement encoding to directly edit the `LimitOrder`'s buy and sell amount fields because we are possibly altering the content of the originally signed order. Of course it would probably make the most sense to directly update the `OrderMetaData` with the relevant inflight execution information, but the solver would at least need to cross check the `LimitOrder` it is given with the corresponding `Order`'s metadata (so not to overfill a partially filled solution). My worry here is that once converted to `LimitOrder` the Order data is dropped.